File: go/1848.go
package main

func getMinDistance(nums []int, target int, start int) int {
	best := 10001

	for idx, num := range nums {
		if num == target {
			diff := idx - start
			if diff < 0 {
				diff *= -1
			}

			best = min(best, diff)
		}
	}

	return best
}

File: zig/1848.zig
const std = @import("std");

fn solution(nums: []const i32, target: i32, start: usize) usize {
    var best: usize = 10001;

    for (nums, 0..) |num, idx| {
        if (num == target) {
            best = @min(best, @abs(idx - start));
        }
    }

    return best;
}

const expect = std.testing.expectEqual;
